sharktank/sharktank/layers/rotary_embedding.py

```diff
@@ -81,12 +91,15 @@ def apply_batched_mask(
         xk_out = torch.view_as_real(xk_ * mask).flatten(3)
         return xq_out.type_as(xq), xk_out.type_as(xk)
 
-    @staticmethod
     def _create_rotary_embed_table(
-        max_seqlen: int, dim: int, theta_value: float = 10000.0
+        self,
+        max_seqlen: int,
+        dim: int,
+        theta_value: float = 10000.0,
     ):
         freqs = 1.0 / (
-            theta_value ** (torch.arange(0, dim, 2)[: (dim // 2)].float() / dim)
+            theta_value
+            ** (torch.arange(0, dim, 2, device=self.device)[: (dim // 2)].float() / dim)
```

Comment:
Gross line break but I assume this is an autoformat choice. In brackets so it should be fine.

Reply:
'tis